### DISCUSSION TITLE
Thread for upcoming changes Draft

### DISCUSSION THREAD

**Discussion by codeperfectplus** [Py-Contributors/support#32] [Ideas]
`Created: 31-10-2022 Last Edit: 04-11-2022`

# Initial draft for upcoming changes plans.

Organization-level overview of ongoing and upcoming development. will move to some tool soon as more maintainers will join.

## 1. Workflow Pipeline

All the changes will be done in the dev branch.
for package creation/documentation(readthedocs) main(stable) branch will be use. It's for stable workflow.

## 2. Documentation Change

> To Change mkdocs(one README file documentation) for readthedocs to Sphinx(multipage extended documentation).

- Mkdocs(single page documentation) :- [Example](https://cybel.readthedocs.io/en/v1.0.2/)
- Sphinx Documentation Example:- [Audiobook](https://audiobook.readthedocs.io/en/latest/)

[Sphinx](https://www.sphinx-doc.org/en/master/index.html) -  Sphinx makes it easy to create intelligent and beautiful documentation.

### Documentation Check Table

| Projects | initial draft | Final version(for spelling mistake,errors) | Remark
| ----- | -----  | -----  | ----|
| Audiobook| ✔️ |  ❌  | |
| Random Profile Generator| ✔️ | ❌   ||
| Cybel | ✔️ | ❌||
| Site-cloner| ❌ | ❌||
| Py-Emailer| ❌|❌||
| Filemover | ✔️ |❌|will write after 1st version||
| Dataset-convertor| ❌| ❌|will write after stable version|
| [Quick DB](https://github.com/py-contributors/quickdb)| ✔️| ❌||

## 3. Test Cases and CI/CD pipeline
>  Extends the test cases and CI/CD

we are extending all the documentation for the Same type for all the projects.

Change Packge Type to Cython.

| Project |  Is Cythonize |
| --- | --- |
| [AudioBook](https://github.com/Py-Contributors/audiobook) | ❌|
| [Dataset-convertor](https://github.com/Py-Contributors/dataset-convertor) | ❌|
| [Random-Profile-Generator](https://github.com/Py-Contributors/RandomProfileGenerator) | ❌|
| [Site-cloner](https://github.com/Py-Contributors/site-cloner) | ❌|
| [FileMoverGUI](https://github.com/Py-Contributors/FileMoverGUI) |❌|
| [QuickDB](https://github.com/Py-Contributors/QuickDB)| ❌|

## 4. File mover

1. Initial PyPI package publishes on PyPI.  https://github.com/Py-Contributors/FileMoverGUI/issues/7
2. Make Necessary changes for CLI development as proposed in the issue.

## 5. Random Profile Generator

1. Data should be generated based on Gender. `fname` file will be rewritten. There will be 2 files for now.
2. Generate Geo Coordinate, should be from the same city.
3. Generate Father/Mother's name as well in the profile section. 
4. Generate fake Credit/Debit Card 
5. with the job title there should be job experience it should match with age.

https://github.com/Py-Contributors/RandomProfileGenerator/issues/44

## 6. AudioBook

1. Create A extra extraction engine for docs/Docx  https://github.com/Py-Contributors/AudioBook/issues/71

## 7. Cybel

1. Upgrade Cybel from 1.7.3 to Cybel 2.0.0 as Discord's major version changed. There are a few new features available in the new API.

## 8. Py-Contributors Website

1. Make the website responsive to mobile 
2. Redesign the website, for more information about the organization. so it will be easy to everyone to join the community and contribute.
3. Update the changes links in the website, like for updated documentation in Project cards.

## 9. Upcoming Projects 2023

1. Voice Authenticator (VA) - A voice authentication system that uses
   the user's voice to authenticate the user. The system will be implemented in Python width
   the help of the deep learning library Keras/pytorch. 

2. Accuracy metrics - implementation of machine learning Accuracy metrics
   in Python. like F1 score, Precision, Recall, Confusion Matrix, ROC curve, AUC, etc.

3. Image Augmentation - implementation of image augmentation for the
   training of the neural network models. it will be implemented in
   numba instead of NumPy. Numba is a just-in-time compiler for python 
   code. it is faster than NumPy and it is easier to use than cython. 
   It will make the code faster.
 
 Please suggest the changes, and I will add them to this draft.
 
 @Py-Contributors/maintainers-team 
 @Py-Contributors/python-support
